drafts/inlinemacros.md

```diff
@@ -80,39 +86,37 @@ by-value parameters and that are not side-effect free paths. Arguments to by-nam
 parameters are left alone.
 
    These transformation are intended to preserve the semantics
-of function applications under inlining: An  function call should have the same
-semantics independently on whether the function was maked `inline` or not.
+of function applications under inlining: A function call should have the same
+semantics with respect to side effects independently on whether the function was made `inline` or not.
+
+   After the rewriting is complete, the result of the rewriting is forcibly upcast
+to the type of the original expression. This mechanism works exactly the same 
+as the mechanism of blackbox macro expansion in Scala 2.10+.
```

Comment:
Just to clarify for someone outside EPFL who'll be reading this. I'm realizing that whitebox macros enable important use cases, and I'm not intending to give up on whiteboxity just yet. However, what I would like to do with the new macro system is to start with something that works 100% robustly and then add stuff as opposed to starting with something that is 100% cool and then thinking how to make it work.